README.md

```diff
@@ -32,6 +32,9 @@ Note that Windmill also designates a `u/` directory for storing code per user on
 
 For information on developing scripts, see the [Windmill Scripts quickstart](https://www.windmill.dev/docs/getting_started/scripts_quickstart).
 
+While it is possible in Windmill to [share common logic across scripts with relative imports](https://www.
+windmill.dev/docs/advanced/sharing_common_logic) we have avoided doing so at this time. This decision aims to keep each script as self-contained as possible, promoting modularity and reducing interdependencies. As a trade-off, there may be some redundancy across common operations, such as writing to a database.
```

Comment:
On one hand I am totally open to re-visiting this in the future.  On the surface it seems silly we don't share common logic already.

On the other hand, I think one of the main reasons for struggle with what's documented at that link is that it ignores that each script creates its own (different) dependency stack.  So any shared code needs to work with a wide range dependencies and/or declare its dependencies.  Short of making it a proper package, there's not an easier way to do that.

This, I think, is worth documenting.

Reply:
Something else I'm curious about, and couldn't find information about in the docs: does the Windmill UI include module files containing common code in the list of Scripts? That might be another reason to keep the status quo. I'd vote against having modules mixed in with the core scripts that our users are operating. (And something I'd want to document.)

Reply:
I think this is the answer, from that link:

> your scripts do not necessarily need to have a main function. If they don't, they are assumed to be shared logic and not runnable scripts.